Marlin/src/module/tool_change.cpp

```diff
@@ -221,6 +221,130 @@
+			#endif
+					  current_position[Z_AXIS] += SWITCHING_TOOLHEAD_SECURITY_RAISE;
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  SERIAL_ECHOLNPGM("(1) Raise Z-Axis ");
+					  if (DEBUGGING(LEVELING)) DEBUG_POS("Moving to Raised Z-Position", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position, planner.max_feedrate_mm_s[Z_AXIS], active_extruder);
+					  planner.synchronize();
+
+			 // STEP 2
+					  current_position[X_AXIS] = placexpos;
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  SERIAL_ECHOLNPAIR("(2) place old tool  ", active_extruder);
+					  if (DEBUGGING(LEVELING)) DEBUG_POS("Moving X SwitchPos", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position, planner.max_feedrate_mm_s[X_AXIS], active_extruder);
+					  planner.synchronize();
+
+					  current_position[Y_AXIS] = SWITCHING_TOOLHEAD_Y_POS - SWITCHING_TOOLHEAD_Y_SECURITY + hotend_offset[Y_AXIS][active_extruder];
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					 (DEBUGGING(LEVELING)) DEBUG_POS("Moving Y SwitchPos + Security", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position, planner.max_feedrate_mm_s[Y_AXIS], active_extruder);
+					  planner.synchronize();
+
+					  // STEP 3
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  SERIAL_ECHOLNPGM("(3) Unlock and Place Toolhead ");
+			#endif
+					  MOVE_SERVO(SWITCHING_TOOLHEAD_SERVO_NR, angles[1]);
+					  safe_delay(500);
+
+					  current_position[Y_AXIS] = SWITCHING_TOOLHEAD_Y_POS + hotend_offset[Y_AXIS][active_extruder];
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  (DEBUGGING(LEVELING)) DEBUG_POS("Moving Y SwitchPos", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position,(planner.max_feedrate_mm_s[Y_AXIS] * 0.5), active_extruder);
+					  planner.synchronize();
+					  safe_delay(200);
+					  current_position[Y_AXIS] -= SWITCHING_TOOLHEAD_Y_CLEAR;
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  (DEBUGGING(LEVELING)) DEBUG_POS("Moving Back Y Clear", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position, planner.max_feedrate_mm_s[Y_AXIS], active_extruder); // move away from docked toolhead
+					  planner.synchronize();
+
+					  // STEP 4
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  SERIAL_ECHOLNPGM("(4) Move to new Tool Head Position");
+			#endif
+					  current_position[X_AXIS] = grabxpos;
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  if (DEBUGGING(LEVELING)) DEBUG_POS("Moving to X new Toolhead", current_position);
+			#endif
+					  planner.buffer_line_kinematic(current_position, planner.max_feedrate_mm_s[X_AXIS], active_extruder);
+					  planner.synchronize();
+					  current_position[Y_AXIS] = SWITCHING_TOOLHEAD_Y_POS - SWITCHING_TOOLHEAD_Y_SECURITY + hotend_offset[Y_AXIS][active_extruder];
+			#if ENABLED(DEBUG_LEVELING_FEATURE)
+					  (DEBUGGING(LEVELING)) DEBUG_POS("Moving Y SwitchPos + Security", current_position);
```

Comment:
Looks like you forgot the `if` here and other debugging lines.

Reply:
Oh yes I see. I will clean up asap. 
Thank you